cpp/packer.cpp

```diff
@@ -0,0 +1,180 @@
+#include "packer.h"
+
+using namespace::facebook;
+
+void Packer::pack(const jsi::Value& value, jsi::Runtime& runtime, mpack_writer_t* writer) {
+    if(value.isString()) {
+
+        mpack_write_cstr(writer, value.getString(runtime).utf8(runtime).c_str());
+        
+    } else if(value.isNumber()) {
+
+        mpack_write(writer, value.getNumber());
+        
+    } else if(value.isBool()) {
+        
+        mpack_write(writer, value.getBool());
+        
+    } else if(value.isNull()) {
+        
+        mpack_write_nil(writer);
+        
+    } else if(value.isUndefined()){
+        
+        mpack_start_bin(writer, 1);
```

Comment:
See here: https://ludocode.github.io/mpack/md_docs_protocol.html at the end of the page. The author is quite adamant we shouldn't use extensions. The way this is built, I don't foresee any difficulty adding a new bin type we'll just read the value we put in the bin in the reader instead of ignoring it.

Reply:
Yeah I saw that but since they implemented support for it anyway I figured they couldn't be that adamant about it 😅

The way you have it, if you want to add support for binary data you'd need to prefix it with a byte that isn't 'u' to signify that it isn't the "undefined" marker. Using the extension would seem a little cleaner to me but not a big deal.

Reply:
Oh sorry, I thought you meant to add another custom type. But yeah adding normal binary data wouldn't be an issue either with a prefix other than u. That's exactly the way extensions work from what I've seen.

Reply:
https://github.com/msgpack/msgpack/issues/206#issuecomment-386066548

Reply:
I was suggesting using an extension, but I agree it doesn't really change much.